From 839a19bfa03408e831da14c149f90e6dbcaed06e Mon Sep 17 00:00:00 2001
From: Enric Balletbo i Serra <eballetbo@iseebcn.com>
Date: Wed, 21 Sep 2011 12:02:30 +0200
Subject: [PATCH 02/44] OMAP3: igep0020: Remove IGEP0030 support from board-igep0020.c file

I think it's better maintain the IGEP0030 support in a separate file,
it's more easy and clear, so we've decided to remove the IGEP0030
support in board-igep0020.c file wainting to introduce a new file
called board-igep0030.c.

Signed-off-by: Enric Balletbo i Serra <eballetbo@iseebcn.com>
---
 arch/arm/mach-omap2/board-igep0020.c |   43 ++++------------------------------
 1 files changed, 5 insertions(+), 38 deletions(-)

diff --git a/arch/arm/mach-omap2/board-igep0020.c b/arch/arm/mach-omap2/board-igep0020.c
index 35be778..82d2001 100644
--- a/arch/arm/mach-omap2/board-igep0020.c
+++ b/arch/arm/mach-omap2/board-igep0020.c
@@ -74,7 +74,6 @@
 
 #define IGEP2_BOARD_HWREV_B	0
 #define IGEP2_BOARD_HWREV_C	1
-#define IGEP3_BOARD_HWREV	2
 
 static u8 hwrev;
 
@@ -82,11 +81,6 @@ static void __init igep2_get_revision(void)
 {
 	u8 ret;
 
-	if (machine_is_igep0030()) {
-		hwrev = IGEP3_BOARD_HWREV;
-		return;
-	}
-
 	omap_mux_init_gpio(IGEP2_GPIO_LED1_RED, OMAP_PIN_INPUT);
 
 	if (gpio_request_one(IGEP2_GPIO_LED1_RED, GPIOF_IN, "GPIO_HW0_REV")) {
@@ -416,9 +410,6 @@ static int igep_twl_gpio_setup(struct device *dev,
 	igep_gpio_leds[3].gpio = gpio + TWL4030_GPIO_MAX + 1;
 #endif
 
-	if (machine_is_igep0030())
-		return 0;
-
 	/*
 	 * REVISIT: need ehci-omap hooks for external VBUS
 	 * power switch and overcurrent detect
@@ -580,17 +571,6 @@ static const struct usbhs_omap_board_data igep2_usbhs_bdata __initconst = {
 	.reset_gpio_port[2] = -EINVAL,
 };
 
-static const struct usbhs_omap_board_data igep3_usbhs_bdata __initconst = {
-	.port_mode[0] = OMAP_USBHS_PORT_MODE_UNUSED,
-	.port_mode[1] = OMAP_EHCI_PORT_MODE_PHY,
-	.port_mode[2] = OMAP_USBHS_PORT_MODE_UNUSED,
-
-	.phy_reset = true,
-	.reset_gpio_port[0] = -EINVAL,
-	.reset_gpio_port[1] = IGEP3_GPIO_USBH_NRESET,
-	.reset_gpio_port[2] = -EINVAL,
-};
-
 #ifdef CONFIG_OMAP_MUX
 static struct omap_board_mux board_mux[] __initdata = {
 	{ .reg_offset = OMAP_MUX_TERMINATOR },
@@ -613,7 +593,7 @@ static void __init igep_wlan_bt_init(void)
 		igep_wlan_bt_gpios[0].gpio = IGEP2_RB_GPIO_WIFI_NPD;
 		igep_wlan_bt_gpios[1].gpio = IGEP2_RB_GPIO_WIFI_NRESET;
 		igep_wlan_bt_gpios[2].gpio = IGEP2_RB_GPIO_BT_NRESET;
-	} else if (hwrev == IGEP2_BOARD_HWREV_C || machine_is_igep0030()) {
+	} else if (hwrev == IGEP2_BOARD_HWREV_C) {
 		igep_wlan_bt_gpios[0].gpio = IGEP2_RC_GPIO_WIFI_NPD;
 		igep_wlan_bt_gpios[1].gpio = IGEP2_RC_GPIO_WIFI_NRESET;
 		igep_wlan_bt_gpios[2].gpio = IGEP2_RC_GPIO_BT_NRESET;
@@ -661,14 +641,10 @@ static void __init igep_init(void)
 	 */
 	igep_wlan_bt_init();
 
-	if (machine_is_igep0020()) {
-		omap_display_init(&igep2_dss_data);
-		igep2_display_init();
-		igep2_init_smsc911x();
-		usbhs_init(&igep2_usbhs_bdata);
-	} else {
-		usbhs_init(&igep3_usbhs_bdata);
-	}
+	omap_display_init(&igep2_dss_data);
+	igep2_display_init();
+	igep2_init_smsc911x();
+	usbhs_init(&igep2_usbhs_bdata);
 }
 
 MACHINE_START(IGEP0020, "IGEP v2 board")
@@ -681,12 +657,3 @@ MACHINE_START(IGEP0020, "IGEP v2 board")
 	.timer		= &omap3_timer,
 MACHINE_END
 
-MACHINE_START(IGEP0030, "IGEP OMAP3 module")
-	.boot_params	= 0x80000100,
-	.reserve	= omap_reserve,
-	.map_io		= omap3_map_io,
-	.init_early	= igep_init_early,
-	.init_irq	= omap3_init_irq,
-	.init_machine	= igep_init,
-	.timer		= &omap3_timer,
-MACHINE_END
-- 
1.7.4.1

